modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java

```diff
@@ -1418,6 +1422,70 @@ public String getFlowchart(String flowchartPath) {
         return flowchartString;
     }
 
+    /**
+     *
+     * @param processName
+     * @param processVersion
+     * @param bpmndesignJson
+     * @return the bpmn diagram json string for a process
+     */
+    public String uploadBpmnDesign(String processName, String processVersion, String bpmndesignJson) {
+        String processId = "NA";
+        log.debug("Creating bpmn design...");
+        try {
+            RegistryService registryService = ProcessCenterServerHolder.getInstance().getRegistryService();
+            if (registryService != null) {
+                UserRegistry reg = registryService.getGovernanceSystemRegistry();
+                Resource bpmnDesignContentResource = reg.newResource();
+                bpmnDesignContentResource.setContent(bpmndesignJson);
+                bpmnDesignContentResource.setMediaType("application/json");
+                String bpmnDesignContentPath = "bpmnDesign/" + processName + "/" + processVersion;
+                reg.put(bpmnDesignContentPath, bpmnDesignContentResource);
+                String processPath = "processes/" + processName + "/" + processVersion;
+
+                // update process by linking the pdf asset
+                Resource processAsset = reg.get(processPath);
+                byte[] processContentBytes = (byte[]) processAsset.getContent();
+                String processContent = new String(processContentBytes);
+                Document processXMLContent = stringToXML(processContent);
+                processXMLContent.getElementsByTagName("bpmnDesign").item(0).getFirstChild().setTextContent(bpmnDesignContentPath);
+
+                String newProcessContent = xmlToString(processXMLContent);
+                processAsset.setContent(newProcessContent);
+                reg.put(processPath, processAsset);
+
+                Resource storedProcessAsset = reg.get(processPath);
+                processId = storedProcessAsset.getUUID();
+            }
+        } catch (Exception e) {
```

Comment:
I have fixed the above issue with commit b17269a. The specific exceptions are catched without catching the generic exception.